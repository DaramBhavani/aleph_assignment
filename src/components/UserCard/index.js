import React from "react";
import { IconButton } from "../index";
import Styled from "styled-components";

const StyledUserCard = Styled.div`
display:flex;
flex-direction:column;
justify-content:center;
border:1px solid black;
box-shadow:0.5px 1px 4px rgba(0, 0, 0, 0.4);
border-radius:.5rem;
max-width:320px;
flex-basis:100%;
margin:1rem;

&:hover{
  transform:translateY(-0.5rem) scale(1);
}
`;

const StyledContent = Styled.div`
padding:1.5rem;
`;
const UserCard = ({ name, email, phone, company }) => {
  return (
    <StyledUserCard data-testid="usercard">
      <StyledContent>
        <p>
          Name: <span>{name}</span>
        </p>
        <p>
          Email: <span>{email}</span>
        </p>
        <p>
          Mobile No: <span>{phone}</span>
        </p>
        <p>
          Company Name: <span>{company.name} </span>
        </p>
      </StyledContent>
      <IconButton iconValue="&#9990;" iconText="Contact Now" />
    </StyledUserCard>
  );
};

export default UserCard;
