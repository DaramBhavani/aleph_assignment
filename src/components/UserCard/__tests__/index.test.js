import React from "react";
import UserCard from "./../index";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

describe("UserCard", () => {
  let data = {
    name: "Leanne Graham",
    email: "Sincere@april.biz",
    phone: 98765555,
    company: {
      name: "Romaguera-Crona",
    },
  };

  it("verify the prop value for name", () => {
    const { getByText, container } = render(<UserCard {...data} />);
    expect(container.querySelector("span").firstChild.textContent).toBe(
      data.name
    );
  });

  it("renders correctly", () => {
    const tree = renderer.create(<UserCard {...data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
