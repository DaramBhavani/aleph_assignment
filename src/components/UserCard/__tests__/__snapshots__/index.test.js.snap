// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard renders correctly 1`] = `
<div
  className="sc-dlfnbm eGOGNR"
  data-testid="usercard"
>
  <div
    className="sc-hKgILt erglPN"
  >
    <p>
      Name: 
      <span>
        Leanne Graham
      </span>
    </p>
    <p>
      Email: 
      <span>
        Sincere@april.biz
      </span>
    </p>
    <p>
      Mobile No: 
      <span>
        98765555
      </span>
    </p>
    <p>
      Company Name: 
      <span>
        Romaguera-Crona
         
      </span>
    </p>
  </div>
  <div>
    <button
      className="sc-bdfBwQ edPwTU"
      data-testid="icon-button"
    >
      <span>
        âœ†
      </span>
      <span>
        Contact Now
      </span>
    </button>
  </div>
</div>
`;
